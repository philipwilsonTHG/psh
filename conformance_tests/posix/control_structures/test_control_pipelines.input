# Test control structures in pipelines
# PSH supports control structures as pipeline components (v0.37.0 feature)

# Test while loop in pipeline
echo -e "apple\nbanana\ncherry" | while read fruit; do
    echo "Processing: $fruit"
done

# Test for loop in pipeline  
echo "1 2 3" | for i in $(cat); do
    echo "Number: $i"
done

# Test if statement in pipeline
echo "test" | if grep -q "test"; then
    echo "Found test in input"
else
    echo "No test found"
fi

# Test case statement in pipeline
echo "hello" | case "$(cat)" in
    hello)
        echo "Greeting detected"
        ;;
    goodbye)
        echo "Farewell detected"
        ;;
    *)
        echo "Unknown message"
        ;;
esac

# Test complex pipeline with multiple control structures
echo -e "1\n2\n3\n4\n5" | while read num; do
    if [ "$((num % 2))" = "0" ]; then
        echo "Even: $num"
    else
        echo "Odd: $num"
    fi
done | while read result; do
    echo "Result: $result"
done