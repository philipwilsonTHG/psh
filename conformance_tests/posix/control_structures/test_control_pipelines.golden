Processing: apple
Processing: banana
Processing: cherry
Number: 1
Number: 2
Number: 3
Greeting detected
Result: Odd: 1
Result: Even: 2
Result: Odd: 3
Result: Even: 4
Result: Odd: 5
